relayer/provider/cosmos/provider.go

```diff
@@ -1128,72 +1137,81 @@ func (cc *CosmosProvider) relayPacketsFromResultTx(ctx context.Context, src, dst
+				return rcvPackets, timeoutPackets, nil
+			}
+		}
 	}
 
 	return nil, nil, fmt.Errorf("no packet data found")
 }
 
 // acknowledgementsFromResultTx looks through the events in a *ctypes.ResultTx and returns
 // relayPackets with the appropriate data
-func acknowledgementsFromResultTx(dstChanId, dstPortId, srcChanId, srcPortId string, res *ctypes.ResultTx) ([]provider.RelayPacket, error) {
+func acknowledgementsFromResultTx(dstChanId, dstPortId, srcChanId, srcPortId string, resp *provider.RelayerTxResponse) ([]provider.RelayPacket, error) {
 	var ackPackets []provider.RelayPacket
-	for _, e := range res.TxResult.Events {
-		if e.Type == waTag {
+
+	rp := &relayMsgPacketAck{pass: false}
+	for key, attrVal := range resp.Events {
+		rp.pass = false
+		eventType, attrKey := splitEventKey(key)
+
+		if eventType == waTag {
 			// NOTE: Src and Dst are switched here
-			rp := &relayMsgPacketAck{pass: false}
-			for _, p := range e.Attributes {
-				if string(p.Key) == srcChanTag {
-					if string(p.Value) != srcChanId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == dstChanTag {
-					if string(p.Value) != dstChanId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == srcPortTag {
-					if string(p.Value) != srcPortId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == dstPortTag {
-					if string(p.Value) != dstPortId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == ackTag {
-					rp.ack = p.Value
+			if attrKey == srcChanTag {
+				if attrVal != srcChanId {
+					rp.pass = true
+					continue
 				}
-				if string(p.Key) == dataTag {
-					rp.packetData = p.Value
+			}
+			if attrKey == dstChanTag {
+				if attrVal != dstChanId {
+					rp.pass = true
+					continue
 				}
-				if string(p.Key) == toHeightTag {
-					timeout, err := clienttypes.ParseHeight(string(p.Value))
-					if err != nil {
-						return nil, err
-					}
-					rp.timeout = timeout
+			}
+			if attrKey == srcPortTag {
+				if attrVal != srcPortId {
+					rp.pass = true
+					continue
 				}
-				if string(p.Key) == toTSTag {
-					timeout, _ := strconv.ParseUint(string(p.Value), 10, 64)
-					rp.timeoutStamp = timeout
+			}
+			if attrKey == dstPortTag {
+				if attrVal != dstPortId {
+					rp.pass = true
+					continue
 				}
-				if string(p.Key) == seqTag {
-					seq, _ := strconv.ParseUint(string(p.Value), 10, 64)
-					rp.seq = seq
+			}
+			if attrKey == ackTag {
+				rp.ack = []byte(attrVal)
+			}
+			if attrKey == dataTag {
+				rp.packetData = []byte(attrVal)
+			}
+			if attrKey == toHeightTag {
+				timeout, err := clienttypes.ParseHeight(attrVal)
+				if err != nil {
+					return nil, err
 				}
+				rp.timeout = timeout
+			}
+			if attrKey == toTSTag {
+				timeout, _ := strconv.ParseUint(attrVal, 10, 64)
+				rp.timeoutStamp = timeout
+			}
+			if attrKey == seqTag {
+				seq, _ := strconv.ParseUint(attrVal, 10, 64)
```

Comment:
If it was okay to return an error from `clienttypes.ParseHeight` on L1194, then it should be okay to check and return the error from these two calls to `strconv.ParseUint`.